test/testdrive/pg-catalog.td

```diff
@@ -145,7 +145,7 @@ name                nullable    type
  rolsuper           true        boolean
  rolinherit         false       boolean
  rolcreaterole      true        boolean
- rolcreatedb        false       boolean
```

Comment:
Is this potentially surprising?

Reply:
Indeed it is. I checked on this, and it is due to a plan regression. The plan became more complex, and we can no longer tell that this column is not nullable.

I've spot checked a few more plans, and they all either improved or didn't change (as indicated by the plans' number of lines):
```
pg_class_all_databases                  114 -> 107 (Seems to be SemijoinIdempotence)
pg_depend				no change
pg_description_all_databases_ind 	no change
pg_index				no change
pg_matviews				60 -> 52   (Seems to be SemijoinIdempotence)
pg_type_all_databases_ind_new	        176 -> 170 (Seems to be SemijoinIdempotence)
table_privileges_old			no change
```

Also, the changes in `mz-arrangement-sharing.td` indicate that plans improved overall.

I'm still looking into what happened in the regressed plan.

Reply:
It seems that we are missing a `RedundantJoin`. I'm not sure why, but might be due to some of the `ProjectionLifting` that is happening after `ReduceElision` elides some `Distinct`s. It's hard to imagine this being some fundamental problem; it's more likely that we'd just need to make `ReduceElision` slightly smarter.

cc @bosconi: This is one of those situations that I mentioned last week where a transform not being "complete" enough creates a plan regression when some minor thing changes in the plan.

Reply:
I'd say we can still merge the PR, because overall there is an improvement.

I added the plan regression to https://github.com/MaterializeInc/materialize/issues/22646, which is already tracking some cases where `RedundantJoin` misses something that it could probably do if it were slightly smarter.